service/history/api/create_workflow_util.go

```diff
@@ -59,7 +59,7 @@ type (
 		RunID            string
 		LastWriteVersion int64
 	}
-	CreateLeaseFunc func(shard.Context, workflow.MutableState) (WorkflowLease, error)
+	CreateLeaseFunc func(WorkflowLease, shard.Context, workflow.MutableState) (WorkflowLease, error)
```

Comment:
As a quick fix, I had to add another parameter here for the case for the lease actually already exists. This is not pretty, but it works.